scripts/helpers/eosio.sh

```diff
@@ -0,0 +1,551 @@
+        && rm -f boost_$BOOST_VERSION.tar.bz2 \
+        && rm -rf $BOOST_LINK_LOCATION"        
+        echo " - Boost library successfully installed @ ${BOOST_ROOT}"
+        echo ""
+    else
+        echo " - Boost library found with correct version @ ${BOOST_ROOT}"
+        echo ""
+    fi
+}
+
+function ensure-llvm() {
+    echo "${COLOR_CYAN}[Ensuring LLVM 4 support]${COLOR_NC}"
+    if [[ ! -d $LLVM_ROOT ]]; then
+        if [[ $ARCH == "Darwin" ]]; then # Handle brew installed llvm@4
+	        execute ln -s /usr/local/opt/llvm@4 $LLVM_ROOT
+	        echo " - LLVM successfully linked from /usr/local/opt/llvm@4 to ${LLVM_ROOT}"
+        else
+            if $PIN_COMPILER || $BUILD_CLANG; then
+                CMAKE_FLAGS="-DCMAKE_INSTALL_PREFIX='${LLVM_ROOT}' -DLLVM_TARGETS_TO_BUILD=host -DLLVM_BUILD_TOOLS=false -DLLVM_ENABLE_RTTI=1 -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE='${BUILD_DIR}/pinned_toolchain.cmake' .."
+            else
+                if [[ $NAME == "Ubuntu" ]]; then
+                    execute ln -s /usr/lib/llvm-4.0 $LLVM_ROOT
+                    echo " - LLVM successfully linked from /usr/lib/llvm-4.0 to ${LLVM_ROOT}"
+                    return 0
+                fi
+                CMAKE_FLAGS="-G 'Unix Makefiles' -DCMAKE_INSTALL_PREFIX=${LLVM_ROOT} -DLLVM_TARGETS_TO_BUILD='host' -DLLVM_BUILD_TOOLS=false -DLLVM_ENABLE_RTTI=1 -DCMAKE_BUILD_TYPE=Release .."
+            fi
+            execute bash -c "cd ${OPT_DIR} \
+            && git clone --depth 1 --single-branch --branch $LLVM_VERSION https://github.com/llvm-mirror/llvm.git llvm && cd llvm \
+            && mkdir build \
+            && cd build \
+            && ${CMAKE} ${CMAKE_FLAGS} \
+            && make -j${JOBS} \
+            && make install"
+            echo " - LLVM successfully installed @ ${LLVM_ROOT}"
+            echo ""
+        fi
+    else
+        echo " - LLVM found @ ${LLVM_ROOT}."
+        echo ""
+    fi
+}
+
+
+function build-clang() {
+    if $BUILD_CLANG; then
+        echo "${COLOR_CYAN}[Ensuring Clang support]${COLOR_NC}"
+        if [[ ! -d $CLANG_ROOT ]]; then
+            execute bash -c "cd ${TEMP_DIR} \
+            && rm -rf clang8 \
+            && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/llvm.git clang8 \
+            && cd clang8 && git checkout $PINNED_COMPILER_LLVM_COMMIT \
+            && cd tools \
+            && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/lld.git \
+            && cd lld && git checkout $PINNED_COMPILER_LLD_COMMIT && cd ../ \
+            && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/polly.git \
+            && cd polly && git checkout $PINNED_COMPILER_POLLY_COMMIT && cd ../ \
+            && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/clang.git clang && cd clang/tools \
+            && git checkout $PINNED_COMPILER_CLANG_VERSION \
```

Comment:
This line and the one above should instead be:
```
            && git clone --single-branch --branch $PINNED_COMPILER_BRANCH https://git.llvm.org/git/clang.git clang && cd clang \
            && git checkout $PINNED_COMPILER_CLANG_COMMIT && cd tools \
```

Reply:
Haven't tested it out, but I believe this change should resolve the following issue: https://github.com/EOSIO/eos/issues/7472